Documentation cleanup and rename Ternary to ConditionalOperator (#1066)
hyperledger/solang: docs/language/expressions.rst

Increment and Decrement operators
_________________________________

> The post-increment and pre-increment operators are implemented like you would expect. So, ``i++``
> evaluates to the value of ``i`` before incrementing, and ``++i`` evaluates to value of ``i``

Reviewer: ```suggestion
The post-increment and pre-increment operators are implemented by reading the variable's value before modifying it. ``i++``
evaluates to the value of ``i`` before incrementing, and ``++i`` evaluates to value of ``i``
```

`like you would expect` is not a clear way to say this. People have different understanding (sometimes wrong) about how things work.

in the EIP.

The contract code for a contract, i.e. the binary WebAssembly or BPF, can be retrieved using the

> ``type(c).creationCode`` and ``type(c).runtimeCode`` fields, as ``bytes``. On evm,

Reviewer: Aren't we using EVM in upper case everywhere?

> the constructor code is in the ``creationCode`` WebAssembly and all the functions are in

Reviewer: Why `creationCode` WebAssembly is the constructor for EVM, if we does not uses WASM?

Author: This does not make much sense, I've re-written it.

> the ``runtimeCode``. Both Parity Substrate and Solana use the same
> code for both, so both fields will evaluate to the same value.

Reviewer: ```suggestion
the ``runtimeCode``. Parity Substrate and Solana use the same
code for both, so those fields will evaluate to the same value.
```
`both` was repeated three times.

_______

Solidity is very strict about the sign of operations, and whether an assignment can truncate a

> value. You can force the compiler to accept truncations or differences in sign by adding an

Reviewer: ```suggestion
value. You can force the compiler to accept truncations or sign shifts by adding an
```
`Differences in sign` sounds awkward to me.

Author: I don't think `sign shifts` is very clear. I've changed it to `sign changes`
